aio/content/examples/universal/src/app/absolute-url.interceptor.ts

```diff
@@ -0,0 +1,34 @@
+// #docplaster
+// #docregion
+import {HttpHandler, HttpInterceptor, HttpRequest} from '@angular/common/http';
+import {Inject, Injectable, Optional} from '@angular/core';
+
+// #enddocregion
+// NOTE:
+//
+// This interceptor is not needed in this example app, because the app uses
+// `@nguniversal/express-engine`, which takes care of converting URLs to absolute automatically.
+// Such an interceptor would only be needed if the app did not use an `@nguniversal/*-engine`
+// package.
+//
+// The interceptor will have no effect here, because
+// `SOME_TOKEN_UNDER_WHICH_THE_FULL_URL_IS_PROVIDED` is not provided and thus `fullUrl` will be
+// `null`.
+
+// #docregion
+@Injectable()
+export class AbsoluteUrlInterceptor implements HttpInterceptor {
+  constructor(
+      @Inject('SOME_TOKEN_UNDER_WHICH_THE_FULL_URL_IS_PROVIDED') @Optional()
+      private fullUrl: URL | null) {}
+
+  intercept(req: HttpRequest<any>, next: HttpHandler) {
```

Comment:
```suggestion
  intercept(req: HttpRequest<unknown>, next: HttpHandler) {
```

Can you please use `unknown` whenever we can instead of `any` as `any` can lead to lots of type inference issues... thanks!

Reply:
I just move the code snippet from the guide to a file, so I kept the original code. I assume `any` was used to match the [HttpInterceptor#intercept()][1] signature (which features `any`).

In any case, this code is removed in the following fixup commit, so this doesn't really matter :grin:
I'll merge the fixup commit into this one to avoid confusion.

[1]: https://angular.io/api/common/http/HttpInterceptor#intercept